src/python/pants/base/worker_pool.py

```diff
@@ -36,12 +36,20 @@ class WorkerPool(object):
   may not be effective. Use this class primarily for IO-bound work.
   """
 
-  def __init__(self, parent_workunit, run_tracker, num_workers):
+  def __init__(self, parent_workunit, run_tracker, num_workers, thread_name_prefix):
```

Comment:
Because you always use the workunit name as the prefix, we could probably not add this argument, not change all of the callers, and instead just use `parent_workunit.name` in this function where you're currently using `thread_name_prefix`?

Reply:
For `parent_workunit` we use `workunit.parent` and for `parent_workunit.name` I used `workunit.name`. A solution could be to pass just the `workunit`, but `background_worker_pool` as `parent_workunit` takes `self.get_background_root_workunit()` that doesn't have a parent and `background_worker_pool` threads are not created within workunit context.
So it could be better to stay with the current solution.